### PR TITLE
Correct default Nextflow version for Platform version 23.3.0

### PR DESCRIPTION
**Pull request by drpatelh**
- Correct default Nextflow version for Platform version 23.3.0
- Remove duplicate entry in functionality matrix

### CODE REVIEW

File: platform_versioned_docs/version-23.4.0/functionality_matrix/functionality_matrix.mdx [hunk 1]
```diff
@@ -19,7 +19,6 @@ nf-launcher versions prefixed with j17 refer to Java version 17; j11 refers to J
 | 23.4.1           | j17-23.10.1         | 23.10.1          | 2.2            |
 | 23.4.0           | j17-23.04.3         | 23.04.3          | 2.1            |
 | 23.3.0           | j17-23.04.3         | 23.04.3          | 2.1            |
```
Comment (drpatelh) [inline platform_versioned_docs/version-23.4.0/functionality_matrix/functionality_matrix.mdx:21]:
```suggestion
| 23.3.0           | j17-23.10.0         | 23.04.3          | 2.1            |
```

File: platform_versioned_docs/version-23.4.0/functionality_matrix/functionality_matrix.mdx [hunk 2]
```diff
@@ -18,8 +18,7 @@ nf-launcher versions prefixed with j17 refer to Java version 17; j11 refers to J
 | 23.4.2           | j17-23.10.1         | 23.10.1          | 2.2            |
 | 23.4.1           | j17-23.10.1         | 23.10.1          | 2.2            |
 | 23.4.0           | j17-23.04.3         | 23.04.3          | 2.1            |
-| 23.3.0           | j17-23.04.3         | 23.04.3          | 2.1            |
-| 23.3.0           | j17-23.04.3         | 23.04.3          | 2.1            |
+| 23.3.0           | j17-23.10.0         | 23.04.3          | 2.1            |
```
Comment (drpatelh) [inline platform_versioned_docs/version-23.4.0/functionality_matrix/functionality_matrix.mdx:21]:
```suggestion
| 23.3.0           | j17-23.10.0         | 23.10.0          | 2.1            |
```

**Comment by gwright99** [comment]
@drpatelh - the `v23.4.0` line should be purged too - there isn't an image set available for that specific combination.

**Review by pditommaso — APPROVED** [review]

**Comment by drpatelh** [comment]
>  @drpatelh - the v23.4.0 line should be purged too - there isn't an image set available for that specific combination.

Maybe that warrants a separate PR?

**Pull request merged**